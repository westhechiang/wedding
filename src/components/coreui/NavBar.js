import styled from 'styled-components';
import { theme } from '../../ui';

const top = `
  height: 58px;
  background: ${theme.color.blue8};
`;

const bottom = `
  height: 90px;
  background: linear-gradient(
    ${theme.color.white} 35%,
    ${theme.color.gray1} 100%
  );
  border-bottom: 1px solid ${theme.color.gray1};
`;

const sizing = `
  height: 50px;
  background: ${theme.color.white};
`;

const sizingMarket = `
  height: 48px;
  background: ${theme.color.white};
  border-bottom: 1px solid ${theme.color.gray1};
`;

const themes = {
  top,
  bottom,
  sizing,
  sizingMarket,
};

const NavBar = styled.div`
  width: 100%;
  display: block;
  box-sizing: border-box;
  padding: 0 40px;
  position: relative;
  z-index: 1;
  ${props => themes[props.theme]};
`;

export default NavBar;
