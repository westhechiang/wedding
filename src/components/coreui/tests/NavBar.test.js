import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';
import NavBar from '../../../components/coreui/NavBar';

describe('<NavBar />', () => {
  it('should match its empty snapshot', () => {
    const wrapper = shallow(<NavBar />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "top" theme snapshot', () => {
    const wrapper = shallow(<NavBar theme="top" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "bottom" theme snapshot', () => {
    const wrapper = shallow(<NavBar theme="bottom" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "sizing" theme snapshot', () => {
    const wrapper = shallow(<NavBar theme="sizing" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });

  it('should match its "sizingMarket" theme snapshot', () => {
    const wrapper = shallow(<NavBar theme="sizingMarket" />);
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
