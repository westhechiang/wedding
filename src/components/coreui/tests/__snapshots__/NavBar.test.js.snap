// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> should match its "bottom" theme snapshot 1`] = `
<div
  className="sc-bdVaJa jzmFzK"
/>
`;

exports[`<NavBar /> should match its "sizing" theme snapshot 1`] = `
<div
  className="sc-bdVaJa jbKrkc"
/>
`;

exports[`<NavBar /> should match its "sizingMarket" theme snapshot 1`] = `
<div
  className="sc-bdVaJa lcfMjb"
/>
`;

exports[`<NavBar /> should match its "top" theme snapshot 1`] = `
<div
  className="sc-bdVaJa crfISE"
/>
`;

exports[`<NavBar /> should match its empty snapshot 1`] = `
<div
  className="sc-bdVaJa kEBNxY"
/>
`;
